cmd/cmd_util.go

```diff
@@ -0,0 +1,152 @@
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"persistent-volume-migrator/internal/ceph/rbd"
+	"persistent-volume-migrator/internal/kubernetes"
+	logger "persistent-volume-migrator/internal/log"
+
+	v1 "k8s.io/api/core/v1"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+// migratePVC migrates a list of PVCs to CSI.
+func migratePVC(client *k8s.Clientset, pvcs *[]v1.PersistentVolumeClaim) error {
+	var err error
+	for _, pvc := range *pvcs {
+		pv, err := kubernetes.GetPV(client, pvc.Spec.VolumeName)
+		if err != nil {
+			return fmt.Errorf("failed to get PV object with name %s", pvc.Spec.VolumeName)
+		}
+		logger.DefaultLog("PV found %v ", pv)
+		err = kubernetes.UpdateReclaimPolicy(client, pv)
+		if err != nil {
+			return fmt.Errorf("failed to update ReclaimPolicy for PV object %s \n", pvc.Spec.VolumeName)
+		}
+		// Retrive old ceph volume name from PV
+		rbdImageName := kubernetes.GetFlexVolumeName(pv)
+		if rbdImageName == "" {
+			return fmt.Errorf("rbdImageName cannot be empty in Flex PV object: %v", pv)
+		}
+		logger.DefaultLog("flex rbdImageName name %v ", rbdImageName)
+
+		// Delete the PVC object
+		err = kubernetes.DeletePVC(client, &pvc) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+		if err != nil {
+			// TODO have an option to revert back if some Delete fails?
+			return fmt.Errorf("failed to Delete PVC object %s \n", pvc.Name)
+		}
+		// Create new PVC with same name in destination storageclass
+		csiPVC := kubernetes.GenerateCSIPVC(destinationStorageClass, &pvc) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+		logger.DefaultLog("structure of the generated PVC in destination Storageclass \n %v ", csiPVC)
+
+		// Create new CSI PVC
+		csiPV, err := kubernetes.CreatePVC(client, csiPVC, 5)
+		if err != nil {
+			return fmt.Errorf("failed to Create CSI PVC object %s : %v", pvc.Name, err)
+		}
+		logger.DefaultLog("new PVC with same name created in CSI. Persistent Volume: %v ", csiPV)
+
+		// Extract new volume name from CSI PV
+		csiRBDImageName := kubernetes.GetCSIVolumeName(csiPV)
+		if csiRBDImageName == "" {
+			return fmt.Errorf("csiRBDImageName cannot be empty in PV object %v", csiPV)
+		}
+		logger.DefaultLog("csi new volume name: %v ", csiRBDImageName)
+		poolName := kubernetes.GetCSIPoolName(csiPV)
+		if poolName == "" {
+			return fmt.Errorf("poolName cannot be empty in PV object")
+		}
+		logger.DefaultLog("csi poolname: %v ", poolName)
+
+		// Create new connection
+		conn, err := createClusterConnection(client, csiPV)
+		if err != nil {
+			return fmt.Errorf("failed to get cluster config %v \n", err)
+		}
+		defer func() {
+			err = conn.Destroy()
+			if err != nil {
+				logger.ErrorLog("failed to destroy the connection: %v", err)
+			}
+		}()
+
+		// Delete the CSI volume in ceph cluster
+		err = conn.RemoveVolumeAdmin(poolName, csiRBDImageName)
+		if err != nil {
+			return fmt.Errorf("err in RemoveVolume: %v", err)
```

Comment:
better error message?

Reply:
Updated the error messages to make them more descriptive and clear.